### PR TITLE
SCRUM-81-Driver-Orders-FE-Reusable-Footer

### PR DESCRIPTION
**Pull request by asdelay**
## Describe your changes

- Created main page with outlet and bottom menu with animations which will be reused among the application
- Created simple example of navigation through the app using bottom menu
- Added new icons
- Added new “Gray" color to constants
- Deleted box-sizing: border box rule from global css file
- Added new redux slice for monitoring selected page among the app
- Modified svgs rule in vite.config
- Installed newer version of svgr package

## Issue ticket code (and/or) and link

- [SCRUM-81](https://codecraftersintership.atlassian.net/browse/SCRUM-81)

### **General**
Screenshot 2024-11-26 at 19 20 17

https://github.com/user-attachments/assets/b8a1ec07-271f-43ba-a60d-67ca15cb4735

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/pages/MainPage/components/FooterNavigation/index.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  IconsContainer,
+  BarContainer,
+  BarCap,
+  BarMiddle,
+  BarCircle,
+  FirstBarSide,
+  SecondBarSide,
+  ThirdBarSide,
+  FourthBarSide,
+} from './styles';
```
Comment (lyudad) [inline src/pages/MainPage/components/FooterNavigation/index.tsx:14]: not right order of import

File: src/pages/MainPage/components/FooterNavigation/styles.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+  transition: flex-grow ${transitionDuration} ease;
+`;
+export const SecondBarSide = styled.div<FirstBarSideProps>`
+  flex-grow: ${({ active }): number =>
+    active === RoutePage || active === OrdersPage ? 0 : 1};
+  background: ${COLORS.gray};
+
+  transition: flex-grow ${transitionDuration} ease;
+`;
```
Comment (lyudad) [inline src/pages/MainPage/components/FooterNavigation/styles.ts:72]: emplty line

File: src/pages/MainPage/components/FooterNavigation/styles.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+export const ThirdBarSide = styled.div<FirstBarSideProps>`
+  flex-grow: ${({ active }): number =>
+    active === MapPage || active === NotificationsPage ? 0 : 1};
+  background: ${COLORS.gray};
+
+  transition: flex-grow ${transitionDuration} ease;
+`;
```
Comment (lyudad) [inline src/pages/MainPage/components/FooterNavigation/styles.ts:78]: empty line

File: src/pages/MainPage/components/FooterNavigation/styles.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+export const FourthBarSide = styled.div<FirstBarSideProps>`
+  flex-grow: ${({ active }): number => (active === NotificationsPage ? 0 : 1)};
+  background: ${COLORS.gray};
+
+  transition: flex-grow ${transitionDuration} ease;
+`;
```
Comment (lyudad) [inline src/pages/MainPage/components/FooterNavigation/styles.ts:83]: empty line

File: src/pages/MainPage/components/FooterNavigation/index.tsx (hunk 1 above)
Comment (asdelay) [reply src/pages/MainPage/components/FooterNavigation/index.tsx:14]: Fixed (added new rule)

File: src/pages/MainPage/components/FooterNavigation/styles.ts (hunk 2 above)
Comment (asdelay) [reply src/pages/MainPage/components/FooterNavigation/styles.ts:72]: Fixed

File: src/pages/MainPage/components/FooterNavigation/styles.ts (hunk 3 above)
Comment (asdelay) [reply src/pages/MainPage/components/FooterNavigation/styles.ts:78]: Fixed

File: src/pages/MainPage/components/FooterNavigation/styles.ts (hunk 4 above)
Comment (asdelay) [reply src/pages/MainPage/components/FooterNavigation/styles.ts:83]: Fixed

File: src/pages/MainPage/components/FooterNavigation/index.tsx (hunk 1 above)
Comment (asdelay) [reply src/pages/MainPage/components/FooterNavigation/index.tsx:14]: Sorry, just to clarify. If you say fix and merge do I have to resend fixed version of PR or I can literally fix and merge?

**Pull request merged**